### PR TITLE
[PYTHON] Add OV Types support to parameter and constant from opsets

### PR DESCRIPTION
**Pull request by jiwaszki**
### Details:
 - `dtype` parameter is now accepting openvino's Type class:
 ```python
 openvino.runtime.opset8.constant(np.arange(9).reshape(3, 3), Type.f32)
 openvino.runtime.opset8.parameter([2, 2, 2], dtype=Type.f32, name="A")
 ```
 - Refactor of tests and error handling.

### Tickets:
 - 79655

### CODE REVIEW

File: src/bindings/python/tests/test_ngraph/test_basic.py [hunk 1]
```diff
@@ -74,6 +76,17 @@ def test_ngraph_function_api():
...
+        Type.f32,
+        Type.f64,
+        Type.i8,
+        Type.i16,
+        Type.i32,
+        Type.i64,
+        Type.u8,
+        Type.u16,
+        Type.u32,
+        Type.u64,
```
Comment (ilyachur) [inline src/bindings/python/tests/test_ngraph/test_basic.py:89]: Can we support `u4`/``i4` and other custom types?

File: src/bindings/python/src/openvino/runtime/opset1/ops.py [hunk 2]
```diff
@@ -1792,11 +1791,13 @@ def pad(
 
 @nameable_op
 def parameter(
-    shape: TensorShape, dtype: NumericType = np.float32, name: Optional[str] = None
+    shape: TensorShape, dtype: Union[NumericType, Type] = np.float32, name: Optional[str] = None
 ) -> Parameter:
     """Return an openvino Parameter object."""
-    element_type = get_element_type(dtype)
-    return Parameter(element_type, PartialShape(shape))
+    return Parameter(get_element_type(dtype)
```
Comment (ilyachur) [inline src/bindings/python/src/openvino/runtime/opset1/ops.py:1797]: Does it work for constant op too?

File: src/bindings/python/src/openvino/runtime/utils/types.py [hunk 3]
```diff
@@ -121,11 +120,11 @@ def get_shape(data: NumericData) -> TensorShape:
     return []
 
 
-def make_constant_node(value: NumericData, dtype: NumericType = None) -> Constant:
+def make_constant_node(value: NumericData, dtype: Union[NumericType, Type] = None) -> Constant:
     """Return an ngraph Constant node with the specified value."""
```
Comment (mitruska) [inline src/bindings/python/src/openvino/runtime/utils/types.py:124]:
```suggestion
    """Return an openvino Constant node with the specified value."""
```

File: src/bindings/python/src/openvino/runtime/opset1/ops.py (hunk 2 above)
Comment (mitruska) [reply src/bindings/python/src/openvino/runtime/opset1/ops.py:1797]: I can see update for `make_constant_node` helper, but there is also a separate function to create `constant` op in opset1/ops.py file, which could be updated to `dtype: Union[NumericType, Type]`:

https://github.com/openvinotoolkit/openvino/blob/6b102784d672a9b020745d937146be0a6f658e27/src/bindings/python/src/openvino/runtime/opset1/ops.py#L315-L323

Comment (jiwaszki) [reply src/bindings/python/src/openvino/runtime/opset1/ops.py:1797]: Yes it works, just need to update hints for types:)

File: src/bindings/python/tests/test_ngraph/test_basic.py (hunk 1 above)
Comment (jiwaszki) [reply src/bindings/python/tests/test_ngraph/test_basic.py:89]: Refer to ticket **57071**.

**Review by mitruska — APPROVED** [review]

**Pull request merged**